<launch>
  <arg name="mav_name" default="$(env HOSTNAME)" />
  <arg name="bag_base" default="/home/$(env USER)/bags/" />
  <arg name="record_cam" default="True" />

  <arg name="cam_topics" value="
  /$(arg mav_name)/bfly/synced_throttle/image_raw
  /$(arg mav_name)/bfly/synced_throttle/camera_info
  "
  if="$(arg record_cam)"/>
  <arg name="cam_topics" value=""
  unless="$(arg record_cam)"/>

  <group ns="$(arg mav_name)">
    <!-- Rosbag  -->
    <node pkg="rosbag" type="record" name="rosbag_record_sensors" required="true" output="screen"
      args="--buffsize=0
      --output-prefix=$(arg bag_base)sensors
      $(arg cam_topics)
      /$(arg mav_name)/rtc
      /$(arg mav_name)/time_sync
      /$(arg mav_name)/lidar_lite/data
      /$(arg mav_name)/us_d1/data
      /$(arg mav_name)/position_controller/pos_yaw_ref
      /$(arg mav_name)/mav_state_estimator/acc_bias
      /$(arg mav_name)/mav_state_estimator/gyro_bias
      /$(arg mav_name)/mav_state_estimator/prediction
      /$(arg mav_name)/mav_state_estimator/odometry
      /$(arg mav_name)/mav_state_estimator/solveThreaded
      /$(arg mav_name)/mav_state_estimator/position_antenna
      /$(arg mav_name)/mav_state_estimator/attitude_antenna
      /$(arg mav_name)/adis16448bmlz/imu
      /$(arg mav_name)/velodyne_packets
      /$(arg mav_name)/velodyne_points
      /$(arg mav_name)/piksi/position_receiver_0/sbp/base_pos_ecef
      -e
      '/$(arg mav_name)/command(.*)|/$(arg mav_name)/dji_sdk(.*)|/$(arg mav_name)/piksi(.*)/ros/(.*)|/$(arg mav_name)/fm_altitude_estimator(.*)'" />

  </group>

</launch>
